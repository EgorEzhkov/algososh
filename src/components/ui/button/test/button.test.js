import renderer from "react-test-renderer";
import { Button } from "../button";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Корректное отображение компонента Button", () => {
  it("Button правильно отображается с текстом", () => {
    const button = renderer.create(<Button text="Пупупу"></Button>).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button правильно отображается без текста", () => {
    const button = renderer.create(<Button></Button>).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button правильно отображается при блокировки кнопки", () => {
    const button = renderer.create(<Button disabled={true}></Button>).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button правильно отображается с индикацией загрузки", () => {
    const button = renderer.create(<Button isLoader={true}></Button>).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button правильно отображается при вызове колбека при клике", () => {
    window.alert = jest.fn();
    render(<Button text="Нажми" onClick={() => alert("пупупу")}></Button>);
    const button = screen.getByText("Нажми");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("пупупу");
  });
});
