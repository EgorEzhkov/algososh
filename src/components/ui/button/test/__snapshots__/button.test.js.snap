// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Корректное отображение компонента Button Button правильно отображается без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Корректное отображение компонента Button Button правильно отображается при блокировки кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Корректное отображение компонента Button Button правильно отображается с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Корректное отображение компонента Button Button правильно отображается с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Пупупу
  </p>
</button>
`;
